crud/src/main/java/com/redhat/lightblue/mediator/QueryPlanNodeExecutor.java

```diff
@@ -224,23 +267,23 @@ private void execute(OperationContext ctx,
         // note the response is not used, but find method changes the supplied context.
         finder.find(nodeCtx,findRequest);
         LOGGER.debug("execute {}: storing {} documents", node.getName(),nodeCtx.getDocuments().size());
+
         for(DocCtx doc:nodeCtx.getDocuments()) {
             DocId id=docIdx.getDocId(doc.getOutputDocument());
-            if(documentCache!=null) {
-                JsonDoc jdoc=documentCache.get(id);
-                if(jdoc==null) {
-                    jdoc=doc.getOutputDocument();
-                    documentCache.put(id,jdoc);
-                }
-            }
             ResultDoc resultDoc=new ResultDoc(doc.getOutputDocument(),id,node);
-            if(parents!=null) {
-                for(ChildDocReference parent:parents) {
-                    resultDoc.setParentDoc(parent.getDocument().getQueryPlanNode(),parent);
-                    parent.getChildren().add(resultDoc);
-                    LOGGER.debug("Linked parent ref:{}",parent);
+
+            if(parents!=null&&!parents.isEmpty()) {
+                for(DocReference parent:parents) {
+                    if(parent instanceof ChildDocReference) {
+                        QueryPlanNode parentNode=parent.getDocument().getQueryPlanNode();
```

Comment:
Line "QueryPlanNode parentNode=parent.getDocument().getQueryPlanNode();" can be removed.